rpmio/rpmkeyring.cc

```diff
@@ -71,6 +71,19 @@ rpmKeyring rpmKeyringFree(rpmKeyring keyring)
     return NULL;
 }
 
+size_t rpmKeyringSize(rpmKeyring keyring, int count_subkeys)
+{
+    if (!keyring) return 0;
+    rdlock lock(keyring->mutex);
+
+    size_t size = 0;
+    for (auto &pair : keyring->keys) {
+	if (count_subkeys or !pair.second->primarykey)
```

Comment:
I didn't even know that `or` was legit c++ :flushed: I dunno what to think about that - it does look's weird and inconsistent in a codebase using `||` everywhere else.

Dunno if this kind of thing comes more naturally to you but I find this condition quite non-obvious and requiring extra processing, and that's never a good thing if it's easily avoidable. I'd write this something like
```
if (count_subkeys && pair.second->primarykey != NULL)
    continue;
++size;
```

That makes clear what primarykey is (ie, it's not a boolean that tells you whether this key is primary key or not, but almost the opposite) and does away with the non-obvious "or not".

Reply:
Your suggestion uses `&&` in place of `or` (instead of `||`).  Was that intentional?

Reply:
It is yeah: "if counting subkeys and this is not a subkey, continue"

Reply:
...and now I'm wondering whether I actually got that logic right :laughing: Need more coffee...

Reply:
Turns out `and` and `or` are totally legal operators in C++. They are different to both the `&` and `&&` variants in that they are boolean and not binary but they are not short circuit but true operators that keep things as a true expression.

Still general consensus seems to be to not use them unless your audience is hard core Python programmers...